src/weapon.py

```diff
@@ -36,13 +34,15 @@ class Beam(Weapon):
     """
 
     beam_projection: arcade.Sprite
+    button_held: bool
```

Comment:
Added to `Beam` which is the only weapon that needs this.  Maybe the name should be changed from `button_held` to `beam_active` or something.

Reply:
I believe with the button functionality you added, we shouldn't need `button_held` at all. Either way, this is functional and I can play with it next time I branch.